lib/Dancer/Plugin/SimpleCRUD.pm

```diff
@@ -610,10 +610,27 @@ CONFIRMDELETE
             my $dbh = database($args{db_connection_name});
             my $where = _get_where_filter_from_args(\%args);
             $where->{$key_column} = $id;
-            $dbh->quick_delete($table_name, $where)
+
+            my %params = params;
+            my $meta_for_hook = {
+                args => \%args,
+                params => \%params,
+                table_name => $table_name,
+                key_column => $key_column,
+            };
+            # fire the pre-delete hook, in case user wants to perform trickery before the delete
+            execute_hook('delete_row_pre_delete', $meta_for_hook );
```

Comment:
Yeah, I agree that it makes sense to make it just like the other pre_* hooks, and worry about whether they should be able to block the action (and if so, how) seperately.

And yeah, the "changes approved" thing was me testing Github'ss new reviews features :)